eduwise-ai/larazen
# Refactoring the CI workflow

## Description
This pull request refactors the CI workflow into three separate jobs: "Test", "Build", and "Deploy". This will allow for better organization and management of the different stages of the CI process.

## Changes Made
- Split the original workflow into three separate jobs: "Test", "Build", and "Deploy".
- Moved the steps related to building and starting the Docker container to the "Test" job.
- Moved the steps related to cache management and building the Docker image to the "Build" job.
- Moved the steps related to deploying the Docker image to the "Deploy" job.

## Benefits
- Improved organization and management of the CI process.
- Faster feedback on test results.
- Faster build times due to cache management.

## How to Test
1. Verify that the CI workflow is triggered for pushes to the `master` and `develop` branches and pull requests to the same branches.
2. Verify that the "Test" job runs the code quality checks and starts and stops the Docker container correctly.
3. Verify that the "Build" job correctly restores and saves the cache, and configures Docker to use GitHub Container Registry.
4. Verify that the "Deploy" job correctly sets and uses the environment variables, and logs in and pushes the Docker image to GitHub Container Registry.

## Screenshots (if applicable)
N/A

## Types of changes
- Refactoring of existing code

## Checklist
- [x ] I have tested my changes locally
- [x ] I have added tests to cover my changes
